packages/lib/EventDispatcher.ts

```diff
@@ -0,0 +1,71 @@
+type Listener<Value> = (data: Value)=> void;
+
+/**
+ * Sends notifications/allows listening for events.
+ *
+ * @param EventKeyType Possible identifiers for events.
+ * @param EventType Type of data associated with events.
+ */
+export default class EventDispatcher<EventKeyType extends string|symbol|number, EventType> {
+	// Partial marks all fields as optional. To initialize with an empty object, this is required.
+	// See https://stackoverflow.com/a/64526384
+	private listeners: Partial<Record<EventKeyType, Array<Listener<EventType>>>>;
+	public constructor() {
+		this.listeners = {};
+	}
+
+	/**
+	 * Sends a notification to all listeners subscribed to the given [notificationKind].
+	 * @param eventType Type of the notification.
+	 * @param value Data associated with the notification.
+	 */
+	dispatch(eventName: EventKeyType, event: EventType = null) {
```

Comment:
Always explicitly mark methods as public or private in new code (again one of those things where adding an eslint rules would be too much troubles due to old code).

Reply:
Good catch! This should be fixed.